### ISSUE TITLE
Determine user weights at runtime

### ISSUE THREAD

**Issue by nathan-beam** [locustio/locust#1781]
### Is your feature request related to a problem? Please describe.

My locust environment requires a high degree of configurability. One parameter that needs to adjusted fairly regularly is the user weights. However, this isn't currently possible without modifying the weight value on the user class itself.

### Describe the solution you'd like

These weights are currently determined by static weight values on the user classes, but it'd be nice to be able to set these dynamically via, say, command line parameters or environment variables.  Since the user classes can already be specified on the command line, that could be expended to include an optional weight attribute:
```
locust UserClass:5 OtherUserClass:1
 ```

### Describe alternatives you've considered

My current workaround is to build a new version of the classes on `init` and set the weight to a custom command line flag. 
```python
@events.init.add_listener
def on_locust_init(environment, **_kwargs):
    user1 = UserClass1
    user2 = UserClass2
    user1.weight = environment.parsed_options.user1_weight
    user2.weight = environment.parsed_options.user1_weight
    environment.user_classes = [ user1 , user2 ]
```

This works fine but it's not the nicest solution. 

### Additional context

I'm available to give this a shot, but I wanted to make sure A) it has value and B) the best implementation from those who are more experienced on this project.

**Comment by mboutet** [comment]
Hi @nathan-beam!

> but it'd be nice to be able to set these dynamically via, say, command line parameters or environment variables

It should already be possible to specify the weight via an environment variable. For example:

```python
class UserClass:
    weight = int(os.getenv("USER_CLASS_WEIGHT", 1))
```

Would that work?

**Comment by nathan-beam** [comment]
This is good, but doesn't solve a couple of edge cases that are kind of unique to my requirements. For example, my application needs to be configurable not just from environment variables, but the command line and a custom version of the web ui as well. Therefore it might require that weighting itself takes place at the `test_start` event rather than the `init` event. Again, I understand this is a pretty unique requirement so it may not be worth the effort

**Comment by mboutet** [comment]
Ok, so what you need is setting/updating the user weights after locust is started. In other words, you could have a situation where you just ran a load test, then decide to update the weights and start a new test without quitting and re-starting locust in between. Is that it?

**Comment by nathan-beam** [comment]
Yeah, that's a good summary

**Comment by mboutet** [comment]
Ok, so the only way I see to hot-reload weights without restarting the locust would be through the web UI (or more specifically through a new flask endpoint). Upon reloading, locust would update the weights of the user class.

However, the distributed case would require the master to send a message to all workers in order for them to update the weights. If/when https://github.com/locustio/locust/pull/1621 is hopefully merged, then only the master would need to reload the weights which should be a little simpler.

In summary, neither command line parameters nor environment variables would allow making a hot-reload of the weights because a restart of locust is required for the new weights to take effect. I don't think it would be too complicated to implement. I would personally wait for https://github.com/locustio/locust/pull/1621 to be incorporated into the master so that we don't need to notify all the workers of the new weights.

**Comment by nathan-beam** [comment]
I've got a pretty decent workaround until (and if) this gets resolved:

```python
from locust import events, HttpUser, task

@events.init_command_line_parser.add_listener
def on_locust_command(parser, **_kwargs):
    parser.add_argument('--user-class-1-weight', env_var='USER_CLASS_1_WEIGHT', type=int, help='Weight for user class 1')
    parser.add_argument('--user-class-2-weight', env_var='USER_CLASS_2_WEIGHT', type=int, help='Weight for user class 2')


@events.test_start.add_listener
def on_test_start(environment, **_kwargs):
    user_class_1 = UserClass1
    user_class_2 = UserClass2

    user_class_1.weight = environment.parsed_options.user_class_1_weight
    user_class_2.weight = environment.parsed_options.user_class_2_weight
    
    environment.user_classes = [ user_class_1, user_class_2 ]


class UserClass1(HttpUser):
    @task
    def task(self):
        pass

class UserClass2(HttpUser):
    @task
    def task(self):
        pass

```

**Comment by mboutet** [comment]
Great you found a workaround! 

However, isn't it the same as https://github.com/locustio/locust/issues/1781#issuecomment-855980403 (minus the command line args)?

**Comment by nathan-beam** [comment]
Yes, but the command line args were necessary. At a later date I'll also need to be able to set it from the web UI as well, and this will handle that scenario once implemented.
